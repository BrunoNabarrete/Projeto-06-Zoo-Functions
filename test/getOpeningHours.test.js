const getOpeningHours = require('../src/getOpeningHours');

describe('Testes da função getOpeningHours', () => {
  it('Testando o retorno se o parâmetro for vázio', () => {
    expect(getOpeningHours()).toEqual({
      Tuesday: { open: 8, close: 6 },
      Wednesday: { open: 8, close: 6 },
      Thursday: { open: 10, close: 8 },
      Friday: { open: 10, close: 8 },
      Saturday: { open: 8, close: 10 },
      Sunday: { open: 8, close: 8 },
      Monday: { open: 0, close: 0 },
    });
  });
  it('Testando recebendo dois argumentos MONDAY e 09:00-AM', () => {
    expect(getOpeningHours('Monday', '09:00-AM')).toBe('The zoo is closed');
  });
  it('Testando recebendo dois argumentos TUESDAY e 09:00-AM', () => {
    expect(getOpeningHours('Tuesday', '09:00-AM')).toBe('The zoo is open');
  });
  it('Testando recebendo dois argumentos WEDNESDAY e 09:00-PM', () => {
    expect(getOpeningHours('Wednesday', '09:00-PM')).toBe('The zoo is closed');
  });
  it('Testando recebendo dois argumentos Thu e 09:00-AM', () => {
    expect(() => getOpeningHours('Thu', '09:00-AM')).toThrowErrorMatchingSnapshot('The day must be valid. Example: Monday');
  });
  it('Testando recebendo dois argumentos Friday e 09:00-ZM', () => {
    expect(() => getOpeningHours('Friday', '09:00-ZM')).toThrowErrorMatchingSnapshot('The abbreviation must be \'AM\' or \'PM\'');
  });
  it('Testando recebendo dois argumentos Saturday e C9:00-AM', () => {
    expect(() => getOpeningHours('Saturday', 'C9:00-AM')).toThrowErrorMatchingSnapshot('The hour should represent a number');
  });
  it('Testando recebendo dois argumentos Sunday e 09:c0-AM', () => {
    expect(() => getOpeningHours('Sunday', '09:c0-AM')).toThrowErrorMatchingSnapshot('The minutes should represent a number');
  });
  it('Testando recebendo dois argumentos Monday e 13:00-AM', () => {
    expect(() => getOpeningHours('Monday', '13:00-AM')).toThrowErrorMatchingSnapshot('The hour must be between 0 and 12');
  });
  it('Testando recebendo dois argumentos Tuesday e 09:60-AM', () => {
    expect(() => getOpeningHours('Tuesday', '09:60-AM')).toThrowErrorMatchingSnapshot('The minutes must be between 0 and 59');
  });
});
