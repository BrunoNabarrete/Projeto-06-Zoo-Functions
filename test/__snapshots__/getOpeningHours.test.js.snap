// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testes da função getOpeningHours Testando recebendo dois argumentos Friday e 09:00-ZM: The abbreviation must be 'AM' or 'PM' 1`] = `"The abbreviation must be 'AM' or 'PM'"`;

exports[`Testes da função getOpeningHours Testando recebendo dois argumentos Monday e 13:00-AM: The hour must be between 0 and 12 1`] = `"The hour must be between 0 and 12"`;

exports[`Testes da função getOpeningHours Testando recebendo dois argumentos Saturday e C9:00-AM: The hour should represent a number 1`] = `"The hour should represent a number"`;

exports[`Testes da função getOpeningHours Testando recebendo dois argumentos Sunday e 09:c0-AM: The minutes should represent a number 1`] = `"The minutes should represent a number"`;

exports[`Testes da função getOpeningHours Testando recebendo dois argumentos Thu e 09:00-AM: The day must be valid. Example: Monday 1`] = `"The day must be valid. Example: Monday"`;

exports[`Testes da função getOpeningHours Testando recebendo dois argumentos Tuesday e 09:60-AM: The minutes must be between 0 and 59 1`] = `"The minutes must be between 0 and 59"`;
